OpenRA.Mods.Common/Widgets/ConfirmationDialogs.cs

```diff
@@ -37,6 +49,21 @@ public static void PromptConfirmAction(string title, string text, Action onConfi
 				if (onCancel != null)
 					onCancel();
 			};
+
+			if (!string.IsNullOrEmpty(restartText) && restartButton != null)
+			{
+				restartButton.IsVisible = () => true;
+				restartButton.OnClick = () =>
+				{
+					if (onRestart != null)
+						onRestart();
+
+					Ui.CloseWindow();
```

Comment:
These should be pulled outside this function so that this can be kept as a general purpose button.